doc/build-helpers/testers.chapter.md

```diff
@@ -169,15 +169,24 @@ The build will fail if `shellcheck` finds any issues.
 
 Checks that the output from running a command contains the specified version string in it as a whole word.
 
-NOTE: In most cases, [`versionCheckHook`](#versioncheckhook) should be preferred, but this function is provided and documented here anyway. The motivation for adding either tests would be:
+Note: Given the current limitations of Nixpkgs's tooling and CI, `versionCheckHook` fits better than `testers.testVersion` in most of typical situations.
```

Comment:
Hmm this suggests a sort of dissatisfaction that might be opinionated, but I don't have a strong opinion regarding it, so I won't make this comment of mine a blocker.

Reply:
Indirectly [you recognized the limitations of the tooling](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1827418529):

> My opinion is that no one should be blamed for not doing something that **our tooling doesn't allow them to easily do**.

Reply:
> [You recognized the limitations of the tooling](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1827418529), at least indirectly:
> 
> > My opinion is that no one should be blamed for not doing something that **our tooling doesn't allow them to easily do**.

I agree that these are "limitations", that also don't allow us to easily run `passthru.tests` of packages automatically etc., but I don't necessarily think that these limitations should be solved at some point. I'd say that calling this a limitation might even be an overstatement, at least for some tools. It'd be very nice if `nixpkgs-review` would have built `passthru.tests` automatically, but I wouldn't say that ofborg is not that well designed. Not only that, even if `nixpkgs-review` would have run it, I'd still claim that `versionCheckHook` is better, because running `nixpkgs-review` is not mandatory, and silent breakages can still rot in your system.